File: system-test/dotnet/SystemTests/Legacy/Mod04/SmokeTests/System/ShopApiSmokeTest.cs
using Common;
using SystemTests.Legacy.Mod04.Base;
using Xunit;

namespace SystemTests.Legacy.Mod04.SmokeTests.System;

public class ShopApiSmokeTest : BaseClientTest
{
    public override Task InitializeAsync()
    {
        SetUpShopApiClient();
        return Task.CompletedTask;
    }

    [Fact]
    public async Task ShouldBeAbleToGoToShop()
    {
        var result = await _shopApiClient!.Health().CheckHealthAsync();
        result.ShouldBeSuccess();
    }
}

File: system-test/java/src/test/java/com/optivem/shop/systemtest/legacy/mod04/smoke/system/ShopApiSmokeTest.java
package com.optivem.shop.systemtest.legacy.mod04.smoke.system;

import com.optivem.shop.systemtest.legacy.mod04.base.BaseClientTest;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static com.optivem.shop.dsl.common.ResultAssert.assertThatResult;

class ShopApiSmokeTest extends BaseClientTest {
    @BeforeEach
    void setUp() {
        setUpShopApiClient();
    }

    @Test
    void shouldBeAbleToGoToShop() {
        var result = shopApiClient.health().checkHealth();
        assertThatResult(result).isSuccess();
    }
}


